core/client/src/main/java/alluxio/client/file/FileSystemContext.java

```diff
@@ -66,15 +66,30 @@
   }
 
   /**
-   * Acquires a block master client from the block master client pool.
+   * Acquires a file system master client from the file system master client pool.
    *
-   * @return the acquired block master client
+   * @return the acquired file system master client
    */
   public FileSystemMasterClient acquireMasterClient() {
     return mFileSystemMasterClientPool.acquire();
   }
 
   /**
+   * Acquires a file system master client from the file system master client pool. The resource is
+   * {@code Closeable}.
+   *
+   * @return the acquired file system master client resource
+   */
+  public CloseableResource<FileSystemMasterClient> acquireMasterClientResource() {
```

Comment:
I think this can be confusing because the FileSystemMasterClient itself has a close method. I am ok with this change though.